File: HW/hw7/HW7.java
package HW;

import java.util.Arrays;

public class HW7 {

	public int minMeetingRooms(int[][] intervals) {
		int[] start = new int[intervals.length];
		int[] end = new int[intervals.length];

		for (int i = 0; i < intervals.length; i++) {
			start[i] = intervals[i][0];
			end[i] = intervals[i][1];
		}

		Arrays.sort(start);
		Arrays.sort(end);

		int serverCount = 0;
		int serverIterator = 0;

		for (int i = 0; i < start.length; i++) {
			if (start[i] < end[serverIterator])
				serverCount++;
			else
				serverIterator++;
		}
		return serverCount;
	}

	public static void main(String[] args) {
		int[][] intervals = { { 0, 30 }, { 5, 10 }, { 15, 20 } };
		HW7 x = new HW7();
		System.out.println(x.minMeetingRooms(intervals));
	}

}

File: HW/hw7/HW7.py
'''
Created on Feb 26, 2024

@author: alexxavier
'''
from typing import List

class HW6:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        start = [interval[0] for interval in intervals]
        end = [interval[1] for interval in intervals]

        start.sort()
        end.sort()

        server_count = 0
        server_iterator = 0

        for i in range(len(start)):
            if start[i] < end[server_iterator]:
                server_count += 1
            else:
                server_iterator += 1

        return server_count
    

intervals = [[0, 30], [5, 10], [15, 20]]
HW6 = HW6()
print(HW6.minMeetingRooms(intervals))
    

    